packages/password-input/src/PasswordInput/PasswordInput.spec.tsx

```diff
@@ -303,6 +303,21 @@ describe('packages/password-input', () => {
         aria-labelledby="label"
         stateNotifications={[{ notification: 'hi', state: 'error' }]}
       />
+      <PasswordInput
+        data-attribute="data test"
+        aria-labelledby="label"
+        stateNotifications="error"
+      />
+      <PasswordInput
+        data-attribute="data test"
+        aria-labelledby="label"
+        stateNotifications="valid"
+      />
+      <PasswordInput
+        data-attribute="data test"
+        aria-labelledby="label"
+        stateNotifications="warning"
+      />
```

Comment:
I'm not sure if this is the correct behavior. If someone passes `stateNotifications` and uses a custom container, TS should require `aria-describedby`. 
It looks like i never actually added a test to test this but I was able to get the error in `main` with this test.

<img width="740" alt="Screenshot 2024-04-24 at 2 51 40 PM" src="https://github.com/mongodb/leafygreen-ui/assets/5740159/485b8788-7400-482c-b3ac-4b212beacf01">

Does that make sense?

Reply:
@shaneeza yes that makes sense. I think previously ts would error if `aria-describedby` wasn't defined and `stateNotifications` was defined as a `State` value. In this iteration of the password input, ts shouldn't error since the new default will be the `FormFieldFeedback` with the icon + error/success message

This change still retains backwards compatibility for cases that use a custom container with `aria-describedby` or `stateNotifications` with an array of notifications. Do you think we need to make further adjustments to the types and if so, what does that look like?

Reply:
i think that if we no longer require it, then no ones really going to use `aria-describedby`, unfortunately. Do you wanna bring this up during dev sync tomorrow and get some other opinions?

Reply:
@shaneeza If `aria-describedby` isn't defined, the default `FormFieldFeedback` will render in addition to the custom container. The additional default seems like it would provide enough visual feedback for the consumer to understand that `aria-describedby` needs to be used in order to remove the default. I feel good with this approach as-is but can discuss in dev sync if you think it warrants broader discussion

Reply:
I see, that clears things up a bit. Can you make it clear in the README that you need `aria-describedby` to remove the default message?